src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -489,6 +488,24 @@ void MavlinkReceiver::handle_message_command_both(mavlink_message_t *msg, const
 			send_storage_information(cmd_mavlink.param1 + 0.5f);
 		}
 
+	} else if (cmd_mavlink.command == MAV_CMD_REQUEST_MESSAGE) {
+		uint16_t message_id = (uint16_t)(cmd_mavlink.param1 + 0.5f);
```

Comment:
`std::round` is actually nicer instead of the +0.5f hack.

Reply:
I agree, so I changed a few other instances of the same problem nearby.